src/main/java/com/rackspace/jenkins_nodepool/NodePool.java

```diff
@@ -360,88 +362,97 @@ String idForPath(String path) throws NodePoolException {
-            if (done) {
-                try {
-                    // accept here so that if any error conditions occur, the above update logic will automatically re-submit
-                    // the node request:
-                    allocatedNodes = acceptNodes(request);
-                    break;
-                } catch (Exception ex) {
-                    LOG.log(Level.SEVERE, null, ex);
-                }
+        // Success represents is request fulfilled - everything else is a problem.
+        if (request.getState() == RequestState.fulfilled) {
+            try {
+                allocatedNodes = acceptNodes(request);
+
+                // Get allocated nodes from the request and add to Jenkins
+                final NodePoolNode node = allocatedNodes.get(0);
+                final NodePoolSlave nps = new NodePoolSlave(node, getCredentialsId());
+                final Jenkins jenkins = Jenkins.getInstance();
+                jenkins.addNode(nps);
+                LOG.log(Level.INFO, "Added NodePool slave to Jenkins: {0}", nps);
+            } catch (Exception ex) {
+                LOG.log(Level.WARNING, ex.getClass().getCanonicalName() +
+                        " occurred while accepting nodes. Message: " + ex.getLocalizedMessage() + ". Cleaning up.");
+
+                // Remove the failed request
+                requests.remove(request);
+
+                // Cancel the job as well
+                Jenkins.getInstance().getQueue().cancel(task);
             }
+        } else {
+            LOG.log(Level.WARNING, "Request failed waiting for request state:" + RequestState.fulfilled +
+                    ", actual state:" + request.getState() +
+                    ". Provisioning failed for task:" + task.getName() +
+                    " with node label:" + task.getAssignedLabel().getName() +
+                    ". Cleaning up.");
 
-            Thread.sleep(5000);
-            //TODO: Configurable poll interval for instance ACTIVE
-        }
-        if (requestState != RequestState.fulfilled || allocatedNodes == null) {
-            throw new Exception(MessageFormat.format("Failed to provision node for label {0}", task.getAssignedLabel().getName()));
-        }
+            // Remove the failed request
+            requests.remove(request);
 
-        if (allocatedNodes.isEmpty()) {
-            LOG.warning("Nodepool node request fulfilled with empty node list");
-        } else {
-            // *** Get allocated nodes from the request and add to Jenkins
-            final NodePoolNode node = allocatedNodes.get(0);
-            final NodePoolSlave nps = new NodePoolSlave(node, getCredentialsId());
-            final Jenkins jenkins = Jenkins.getInstance();
-            jenkins.addNode(nps);
-            LOG.log(Level.INFO, "Added NodePool slave to Jenkins: {0}", nps);
+            // Cancel the job as well
+            Jenkins.getInstance().getQueue().cancel(task);
```

Comment:
Cleanup works well for the request and the build :)

Reply:
Awesome!